sdk/lib/drivers/sound/mmebuddy/wave/wodMessage.c

```diff
@@ -118,6 +118,24 @@ wodMessage(
             break;
         }
 
+        case WODM_GETVOLUME:
+        {
+            Result = MmeGetVolume(WAVE_OUT_DEVICE_TYPE,
+                                  DeviceId,
+                                  PrivateHandle,
+                                  &Parameter1);
```

Comment:
I'm quite certain there is a bug here and in the other invocation(s) of this MmeGetVolume in a similar context.
Indeed, as for example the MmeGetPosition etc, when the generic `wodMessage()` callback function is invoked by the caller, the `Parameter1` should already be set to the address of the data buffer that will receive the sought value (for example, the volume).

Reply:
Do you mean this `DWORD_PTR Parameter1` should be already a pointer? Judging by the documentation, it's true, so I should to simply cast `PDWORD` to `DWORD_PTR` I think. Then the problem will be solved.

Reply:
Done @HBelusca. See my updated changes. :slightly_smiling_face: 
Maybe I did not stuck on this problem yet, because `GetVolume` functions was not called in my case yet.